packages/nextjs/app/api/challenges/[challengeId]/submit/route.ts

```diff
@@ -72,19 +72,27 @@ export async function POST(req: NextRequest, props: { params: Promise<{ challeng
         try {
           const referrer = user?.referrer || undefined;
           const originalUtmParams = user?.originalUtmParams;
-          await trackPlausibleEvent(
-            PlausibleEvent.CHALLENGE_SUBMISSION,
-            {
-              challengeId,
-              originalReferrer: referrer,
-              originalUtmSource: originalUtmParams?.utm_source,
-              originalUtmMedium: originalUtmParams?.utm_medium,
-              originalUtmCampaign: originalUtmParams?.utm_campaign,
-              originalUtmTerm: originalUtmParams?.utm_term,
-              originalUtmContent: originalUtmParams?.utm_content,
-            },
-            req,
-          );
+
+          try {
+            await trackPlausibleEvent(
+              PlausibleEvent.CHALLENGE_SUBMISSION,
+              {
+                challengeId,
+                originalReferrer: referrer,
+                originalUtmSource: originalUtmParams?.utm_source,
+                originalUtmMedium: originalUtmParams?.utm_medium,
+                originalUtmCampaign: originalUtmParams?.utm_campaign,
+                originalUtmTerm: originalUtmParams?.utm_term,
+                originalUtmContent: originalUtmParams?.utm_content,
+              },
+              req,
+            );
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
+          } catch (e) {
+            console.error(
+              `Error tracking plausible event ${PlausibleEvent.CHALLENGE_SUBMISSION} for user ${userAddress}`,
+            );
+          }
```

Comment:
I think it's better to have a separate `waitUntil` for `trackPlausibleEvent` and also place that `waitUntil` block after the autograder submission `waitUntil`  so that we first enque the autograder request and then enque the analytics

Just tried it out and seems to work as expected with separte waitUntils 🙌

Reply:
Thanks, updated! [335e33c](https://github.com/BuidlGuidl/SpeedRunEthereum-v2/pull/290/commits/335e33cbeaa01337ca7c1f1161131330a3cbdab6)